pkg/comp-functions/functions/vshnredis/pvcresize.go

```diff
@@ -0,0 +1,276 @@
+	}
+	if err != nil {
+		return false, runtime.NewFatalErr(ctx, "failed to read size from release", err)
+	}
+
+	desiredInt, err := getSizeAsInt(comp.Spec.Parameters.Size.Disk)
+	if err != nil {
+		return false, runtime.NewFatalErr(ctx, "cannot parse desired disk size", err)
+	}
+
+	releaseInt, err := getSizeAsInt(releaseSizeValue)
+	if err != nil {
+		return false, runtime.NewFatalErr(ctx, "cannot parse release disk size", err)
+	}
+
+	return desiredInt > releaseInt, nil
+
+}
+
+func getSizeAsInt(size string) (int64, error) {
+	parsedSize, err := resource.ParseQuantity(size)
+	if err != nil {
+		return 0, err
+	}
+	finalSize, _ := parsedSize.AsInt64()
+	return finalSize, nil
+}
+
+func getObservedOrDesiredRelease(ctx context.Context, iof *runtime.Runtime) (*helmv1beta1.Release, error) {
+	r := &helmv1beta1.Release{}
+	err := iof.Observed.Get(ctx, r, redisRelease)
+	if err != nil && err != runtime.ErrNotFound {
+		return nil, fmt.Errorf("cannot get redis release from observed iof: %v", err)
+	}
+
+	if err == runtime.ErrNotFound {
+		err := iof.Desired.Get(ctx, r, redisRelease)
+		if err != nil {
+			return nil, fmt.Errorf("cannot get redis release from desired iof: %v", err)
+		}
+	}
+	return r, nil
+}
+
+func addDeletionJob(ctx context.Context, iof *runtime.Runtime, comp *vshnv1.VSHNRedis) error {
+	ns := iof.Config.Data["controlNamespace"]
+
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      comp.Name + "-sts-deletion-job",
+			Namespace: ns,
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					ServiceAccountName: "sa-sts-deleter",
+					RestartPolicy:      corev1.RestartPolicyNever,
+					Containers: []corev1.Container{
+						{
+							Name:            "sts-deleter",
+							Image:           "bitnami/kubectl",
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Env: []corev1.EnvVar{
+								{
+									Name:  "STS_NAME",
+									Value: "redis-master",
+								},
+								{
+									Name:  "STS_NAMESPACE",
+									Value: getInstanceNamespace(comp),
+								},
+								{
+									Name:  "STS_SIZE",
+									Value: comp.Spec.Parameters.Size.Disk,
+								},
+								{
+									Name:  "COMPOSITION_NAME",
+									Value: comp.Name,
+								},
+							},
+							Command: []string{
+								"bash",
+								"-c",
+							},
+							Args: []string{
+								stsRecreateScript,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	return iof.Desired.PutIntoObject(ctx, job, comp.Name+"-sts-deleter")
+}
+
+func addStsObserver(ctx context.Context, iof *runtime.Runtime, comp *vshnv1.VSHNRedis) error {
+
+	statefulset := &appsv1.StatefulSet{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "StatefulSet",
+			APIVersion: "apps/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "redis-master",
+			Namespace: getInstanceNamespace(comp),
+		},
+	}
+
+	stsBytes, err := json.Marshal(statefulset)
+	if err != nil {
+		return err
+	}
+
+	observer := &xkubev1.Object{
```

Comment:
maybe it's easier to reuse wrapper func I did? I can do release quite fast and it'll be available on master to rebase

Reply:
Yes I was hoping for that actually :)